pkg/storegateway/bucket.go

```diff
@@ -803,6 +814,17 @@ func debugFoundBlockSetOverview(logger log.Logger, mint, maxt, maxResolutionMill
 
 // Series implements the storepb.StoreServer interface.
 func (s *BucketStore) Series(req *storepb.SeriesRequest, srv storepb.Store_SeriesServer) (err error) {
+	defer func() {
+		if err == nil {
+			return
+		}
+		code := codes.Aborted
```

Comment:
i think it kind of depends. i had to do this because some of the errors which were previously sruficing here https://github.com/grafana/mimir/blob/ad526f8a56b3c8b8c9f10d5e8657db119d8962cd/pkg/storegateway/bucket.go#L1023-L1029

are now surfacing here 

https://github.com/grafana/mimir/blob/78ed13fe9d3893ef3b1c7c52e4938053aa020dba/pkg/storegateway/bucket.go#L951-L954

---

so with the collective changes we are removing this case where we return `Unknown`

https://github.com/grafana/mimir/blob/ad526f8a56b3c8b8c9f10d5e8657db119d8962cd/pkg/storegateway/bucket.go#L910-L913

i couldn't find a place which looks for the Unknown code.

---

to answer your question on the querier. In the querier we just ignore the errors unless it's EOF 
https://github.com/grafana/mimir/blob/3592c25153f9245f65eaf76523e623dfccc3a937/pkg/querier/blocks_store_queryable.go#L737-L744

Reply:
Thanks for the details. I also can't see a reason why this change could break any error handling, given the querier is not checking it and `io.EOF` comes from the client, not the server.